### ISSUE TITLE
"AttributeError: 'bool' object has no attribute 'get'" in salt/state.py on python 2

### ISSUE THREAD

**Issue by johnnybubonic** [saltstack/salt#51281]
### Description of Issue/Question
On a CentOS 6.10 minion (using Python 2.7, salt from the salt-latest repo), I get this curious stacktrace:

```Python
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/site-packages/salt/minion.py", line 1607, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/usr/lib/python2.7/site-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/usr/lib/python2.7/site-packages/salt/modules/state.py", line 764, in apply_
        return highstate(**kwargs)
      File "/usr/lib/python2.7/site-packages/salt/modules/state.py", line 1068, in highstate
        orchestration_jid=orchestration_jid)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 3940, in call_highstate
        return self.state.call_high(high, orchestration_jid)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2764, in call_high
        ret = self.call_chunks(chunks)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2129, in call_chunks
        running = self.call_chunk(low, running, chunks)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2563, in call_chunk
        running = self.call_chunk(low, running, chunks)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2459, in call_chunk
        status, reqs = self.check_requisite(low, running, chunks, pre=True)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2268, in check_requisite
        self.reconcile_procs(running)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2204, in reconcile_procs
        proc = running[tag].get('proc')
    AttributeError: 'bool' object has no attribute 'get'
```
It **doesn't** happen on CentOS 7.6 using the salt py3 repo, so I'd guess it's just a python2/3 inconsistency.

It's worth noting, though, that this seemed to happen when calling the [openssh formula](https://github.com/saltstack-formulas/openssh-formula) and adding a host key via `salt.states.ssh_known_hosts.present`.

### Setup
I have no idea which SLS to include since this occurs on a highstate.

### Steps to Reproduce Issue
See above; it seems more like a python 2/3 thing than something apparently reproducible.

### Versions Report

Minion info:
```YAML
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Mar 30 2018, 15:31:59)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 6.10 Final
         locale: UTF-8
        machine: x86_64
        release: 2.6.32-754.el6.x86_64
         system: Linux
        version: CentOS 6.10 Final
```

**Comment by johnnybubonic** [comment]
i do indeed! weirdly enough, with failhard enabled or disabled, i can't seem to reproduce it. the only thing i can think of is a reboot of the minion. i've been working on my states since then, so it's possible i "fixed" it - but that's being blocked by another issue i need to file so i'm not 100% certain. i'm going to close this for now and i'll re-open if i bump into it again once the other blocker i'm experiencing is sorted. sorry!
